specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/mfe.json

```diff
@@ -0,0 +1,1719 @@
+        ],
+        "summary": "Get datastore secrets.",
+        "operationId": "Datastores_ListSecrets",
+        "parameters": [
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "in": "path",
+            "name": "name",
+            "description": "Datastore name.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DatastoreCredentials"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get datastore secrets.": {
+            "$ref": "./examples/listDatastoreSecrets.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/datastores/{name}/setAzureMLDefault": {
+      "post": {
+        "tags": [
+          "Datastore"
+        ],
+        "summary": "Set default datastore.",
+        "operationId": "Datastores_SetAzureMLDefault",
+        "parameters": [
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "in": "path",
+            "name": "name",
+            "description": "Datastore name.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Success"
+          }
+        },
+        "x-ms-examples": {
+          "Set default datastore.": {
+            "$ref": "./examples/setDefaultDatastore.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/datastores/azuremldefault": {
```

Comment:
This has a lot of downsides.  There is no actual datastore named "azuremldefault" (or at least there shouldn't be).  Can you instead add a "$filter=default eq true" filter to the /datastores collection GET instead?

Reply:
Can you elaborate on the downsides?

Using a reserved name was suggested to us in earlier discussions with ARM API folk. We have an existing backend service that we're mapping this to, and it stores info about the default separately from the datastores themselves, which makes implementing that filter option awkward; that's why we settled on this as an alternative that more closely matches our existing non-ARM APIs.

Reply:
RBAC is odd here.  If I only have read access on a specific datastore I won't be able to see the azuremldefault datastore (even if the one I have access to is the default).  Will "azuremldefault" show up in the collection GET response?  How do I know what datastore resource ID is the default since doing a GET on azuremldefault will return an ID ending in /azuremldefault.  This is not how ARM APIs function.

Reply:
Similarly, if I want to make changes to the default I have to figure out which resourceId it is then do a PUT on that datastore.

Reply:
Alright, those issues make sense. Removed the "azuremldefault" endpoint in favor of the collection GET operation.